### DISCUSSION TITLE
Group-averaging response functions?

### DISCUSSION THREAD

**Discussion by smeisler** [PennLINC/qsiprep#347] [General]
Hello,

I was curious what the community thought about having an alternate way of handling response functions. For between-subject comparisons of CSD metrics (e.g. AFD), one usually uses a group-averaged response function. However, the downside is that individual subject tractography will not be accurate for each subject, since the FODS in each voxel are now informed by the group-averaged rather than subject-specific response profiles.

Do you think it would be worth having a group-averaged option, or should one just venture outside of QSIPrep/Recon for that?

Best,
Steven

**Comment by lizannea** [comment]
Hi Steven, 

I was wondering if the code to calculate the response function for a single-subject, has been made available. Further, would it be possible to get the response function without running the full reconstruction pathway? 

Many thanks, 
Lizanne

@civier @davidjwhite_33

**Comment by smeisler** [comment]
Hi @lizannea ,

Using mrtrix3, you can use [`dwi2response`](https://mrtrix.readthedocs.io/en/latest/reference/commands/dwi2response.html). For most purposes the `dhollander` algorithm should work.

Best,
Steven

**Reply by lizannea** [reply]
Hi Steven, 

Thank you! 

I also think it would be great If there is an option to generates a group response function. I've found, mblesac2 has an example on ' https://git.ecdf.ed.ac.uk/jbrl/neonatal-gfactors/-/tree/master/tracts', but would it also be possible without reorienting the qsiprep preprocessed DWI and bval/bvec to the standard FSL orientation? 

Best, 
Lizanne

**Reply by smeisler** [reply]
Hi,

There’s no need to reorient the data to FSL standard orientation. Although, if you want to run TractSeg software on your data, then they will need to be in FSL orientation.

**Reply by lizannea** [reply]
Hi Steven, 

I didn't realize my data needed to be in FSL orientation for TractSeg. Thank you that was very helpful!

**Comment by araikes** [comment]
@smeisler, I think that the option to have a group response would be really good. Current https://github.com/BIDS-Apps/MRtrix3_connectome has that ability and being able to have an option for a group response would be beneficial.

**Comment by mattcieslak** [comment]
This is something I've been thinking about for awhile too. My approach would be to make some custom json files so that only the response estimation is done. You can start with any of the mrtrix workflows and edit them so that they stop after dwi2response. For example, you could use `--recon-spec just_response.json` where `just_response.json` contains

```json
{
  "name": "mrtrix_response",
  "space": "T1w",
  "atlases": [],
  "anatomical": [],
  "nodes": [
    {
      "name": "msmt_csd",
      "software": "MRTrix3",
      "action": "csd",
      "output_suffix": "msmtcsd",
      "input": "qsiprep",
      "parameters": {
        "mtnormalize": true,
        "response": {
          "algorithm": "dhollander"
        },
        "fod": {
          "algorithm": "msmt_csd",
          "max_sh": [
            8,
            8,
            8
          ]
        }
      }
    }
  ]
}
```

Then you'll have response functions for everyone that goes through this workflow.

That's where qsiprep runs out of usefulness at the moment, though, as I don't have a way to read in response functions into a reconstruction workflow yet. If anyone would be interested in implementing this, I'd be very happy to help get them started!

**Reply by lizannea** [reply]
Thank you so much for your quick response @mattcieslak! It ran successfully but I’m looking at individual differences which is why I would like to implement the group-average response function into the reconstruction. Is this functionality in the development pipeline?

**Reply by smeisler** [reply]
In the meantime, you can get the group average function with `responsemean` https://mrtrix.readthedocs.io/en/latest/reference/commands/responsemean.html
